packages/core/lib/Interfaces/Context.ts

```diff
@@ -0,0 +1,34 @@
+import { IUser } from './User';
+
+export interface IContext {
+  tags?: any;
+  extra?: any;
+  user?: IUser;
+}
+
+export function getDefaultContext() {
+  return {};
+}
+
+export function set<T extends IContext, K extends keyof T>(
+  context: T,
+  key: K,
+  value: any
+) {
+  context[key] = value;
+}
+
+export function merge<T extends IContext, K extends keyof T>(
+  context: T,
+  key: K,
+  value: any
+) {
+  if (!context[key]) {
+    set(context, key, {});
+  }
+  if (value === undefined) {
+    delete context[key];
```

Comment:
Are you sure you want this behavior? Usually `merge` or `assign` functions as well as the object spread operator ignore sources that are falsy.

Reply:
So I took this merge function basically from raven-js and I guess we want to keep the same logic.